app/assets/javascripts/ndt_speed_test.coffee

```diff
@@ -1,42 +1,60 @@
 bind_ndt_speed_calculation = ->
-  ndtServer = undefined
-  ndtPort = '3010'
-  ndtProtocol = 'wss'
-  ndtPath = '/ndt_protocol'
-  ndtUpdateInterval = 1000
-
-  getNdtServer = ->
-    xhr = new XMLHttpRequest
-    mlabNsUrl = 'https://mlab-ns.appspot.com/ndt_ssl?format=json'
-    xhr.open 'GET', mlabNsUrl, true
-    xhr.send()
-
-    xhr.onreadystatechange = ->
-      if xhr.readyState == 4
-        if xhr.status == 200
-          ndtServer = JSON.parse(xhr.responseText).fqdn
-          ndtServerIp = JSON.parse(xhr.responseText).ip
-
-          console.log 'Using M-Lab Server ' + ndtServer
-          document.getElementById('submission_hostname').value = ndtServer;
-
-          # Track NDT server used
-          Sentry.setExtra("mlab_server", ndtServer)
-        else
-          console.log 'M-Lab NS lookup failed.'
-
-          # Track NDT server used
-          Sentry.setExtra("mlab_server", 'failed')
-      return
-    return
-
-  getNdtServer()
 
   NDT_meter = new NDTmeter('#ndt-svg')
   $('#start_ndt_test').on 'click', ->
-    NDT_client = new NDTjs(ndtServer, ndtPort, ndtProtocol, ndtPath, NDT_meter, ndtUpdateInterval)
-    NDT_client.startTest()
-    return
+    # Note: rates are in kbits/s, latency is in milliseconds.
+    s2cRate = undefined
+    c2sRate = undefined
+    minRTT = undefined
+    NDT_meter.onstart()
+    ndt7.test {
+      userAcceptedDataPolicy: true
+      downloadworkerfile: '/assets/js/ndt7-download-worker.min.js'
+      uploadworkerfile: '/assets/js/ndt7-upload-worker.min.js'
+    }, {
+      serverChosen: (server) ->
+        console.log 'Using M-Lab Server ', {
+          machine: server.machine
+          locations: server.location
+        }
+        document.getElementById('submission_hostname').value = server.machine
+        Sentry.setExtra("mlab_server", server.machine)
+        NDT_meter.onstatechange('running_s2c')
+      
```

Comment:
It looks like the empty lines in this function have indentation spaces. Could we make them entirely empty instead?